### ISSUE TITLE
vmctl vm-native massively slower since 1.90.0

### ISSUE THREAD

**Issue by damluk** [VictoriaMetrics/VictoriaMetrics#4092]
### Describe the bug

After upgrading my source and target from 1.87.x to 1.90.0, `vmctl vm-native` take massively more time for small amounts of data.

Typical stats for a sync from 1.87.x to 1.87.x:
```
Initing import process to "http://127.0.0.1:8428/victoria/api/v1/import/native":
Total: 179.28 KiB ↖ Speed: ? p/s
Total: 530.32 KiB ↗ Speed: ? p/s
Total: 946.32 KiB ↘ Speed: ? p/s
Total: 1.46 MiB ↙ Speed: 2.12 MiB p/s
2023/04/07 09:48:35 Import finished!
Total: 1.92 MiB ↖ Speed: 2.51 MiB p/s
2023/04/07 09:48:35 Total time: 1.156093432s
```

Now it is:
```
VictoriaMetrics Native import mode
2023/04/07 10:57:48 Initing import process from "..." to "http://127.0.0.1:8428/victoria/api/v1/import/native" with filter
        filter: match[]={__name__!=""}
        start: 2023-04-07T08:56:03Z
2023/04/07 10:57:48 Exploring metrics...
2023/04/07 10:57:48 Requests to make: 576
2023/04/07 10:58:16 Import finished!
2023/04/07 10:58:16 VictoriaMetrics importer stats:
  time spent while importing: 28.60496489s;
  total bytes: 2.4 MB;
  bytes/s: 84.3 kB;
  requests: 576;
  requests retries: 0;
2023/04/07 10:58:16 Total time: 28.6061416s
```

### To Reproduce

Take before and after measurements.

### Version

victoria-metrics-20230407-010036-tags-v1.90.0-0-gb5d18c0d2

**Comment by damluk** [comment]
This goes with a massive increase of  TCP connections between `vmctl` and the source during the import. The number of "Requests to make" really seems to be the number of separate TCP connections. I have counted 600 connections spanned over 10k packets for 3MB of payload.

**Comment by dmitryk-dk** [comment]
Hi @damluk ! Thank you for the issue. It happens because `vmctl` supports the retry backoff policy, and for correct work, we are using a request to `/api/v1/series` for exploring metric names. This request is heavy for a match like `{name!=""}`, we will find a workaround for that case.

**Comment by dmitryk-dk** [comment]
Hi @damluk ! Can you check if the draft PR improves the speed of the migration process?
My results
```bash
./vmctl vm-native --vm-native-src-addr=http://127.0.0.1:8481/select/0/prometheus \
  --vm-native-dst-addr=http://localhost:8428 \
  --vm-native-filter-match='{__name__="vm_http_request_errors_total"}' \
  --vm-native-filter-time-start='2023-03-01T00:00:00Z' \
  --vm-native-filter-time-end='2023-03-31T23:59:59Z'
VictoriaMetrics Native import mode

2023/04/07 18:44:51 Initing import process from "http://127.0.0.1:8481/select/0/prometheus/api/v1/export/native" to "http://localhost:8428/api/v1/import/native" with filter 
        filter: match[]={__name__="vm_http_request_errors_total"}
        start: 2023-03-01T00:00:00Z
        end: 2023-03-31T23:59:59Z
2023/04/07 18:44:51 Exploring metrics...
Found 1 metrics to import. Continue? [Y/n] 
2023/04/07 18:44:52 Requests to make: 1
Requests to make: 1 / 1 [███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100.00%
2023/04/07 18:44:57 Import finished!
2023/04/07 18:44:57 VictoriaMetrics importer stats:
  time spent while importing: 5.782156417s;
  total bytes: 9.6 MB;
  bytes/s: 1.7 MB;
  requests: 1;
  requests retries: 0;
2023/04/07 18:44:57 Total time: 5.782863584s
```
I have used only one metric name because, on this installation, there is a lot of data

**Comment by damluk** [comment]
> Can you check if the draft PR improves the speed of the migration process?

I have tested with `vmctl-20230407-184346-heads-issue-4092-0-g6f471e678-dirty-830f4acc` without improvements (still around 30s).

**Comment by damluk** [comment]
> Hi @damluk ! Can you check the last commit to the PR? My last test showed the next results.
> ...
> I have changed the logic of HTTP client usage, so the number of connections should be decreased. Please check on your setup. Thanks in advance

Yes!

```
2023/04/10 21:18:44 Import finished!
2023/04/10 21:18:44 VictoriaMetrics importer stats:
  time spent while importing: 753.848463ms;
  total bytes: 2.7 MB;
  bytes/s: 3.5 MB;
  requests: 1;
  requests retries: 0;
2023/04/10 21:18:44 Total time: 755.060353ms
```

**Comment by valyala** [comment]
FYI, the bugfix has been included in [vmctl v1.91.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed.
